api/lib/infrastructure/repositories/general-certification-information-repository.js

```diff
@@ -4,24 +4,68 @@ const { NotFoundError } = require('../../domain/errors');
 const GeneralCertificationInformation = require('../../domain/read-models/GeneralCertificationInformation');
 const CertificationIssueReport = require('../../domain/models/CertificationIssueReport');
 
+const FRANCE_INSEE_CODE = '99100';
+
 module.exports = {
 
   async get({ certificationCourseId }) {
     const certificationCourseDTO = await knex('certification-courses')
       .where({ id: certificationCourseId })
-      .orderBy('createdAt')
       .first();
 
     if (certificationCourseDTO) {
       const certificationIssueReportsDTO = await knex('certification-issue-reports')
         .where({ certificationCourseId });
 
+      if (certificationCourseDTO.birthPostalCode) {
+        const candidateBirthCity = await knex('certification-cpf-cities')
+          .select('name', 'postalCode')
+          .where({ postalCode: certificationCourseDTO.birthPostalCode })
+          .first();
+
+        certificationCourseDTO.birthplace = candidateBirthCity.name;
+        certificationCourseDTO.birthPostalCode = candidateBirthCity.postalCode;
+        certificationCourseDTO.birthCountry = 'FRANCE';
+      }
+
+      if (certificationCourseDTO.birthINSEECode) {
+
+        if (_INSEECodeIsForeign(certificationCourseDTO.birthINSEECode)) {
+          const candidateBirthCountry = await knex('certification-cpf-countries')
+            .select('commonName')
+            .where({ code: certificationCourseDTO.birthINSEECode })
+            .first();
+
+          certificationCourseDTO.birthCountry = candidateBirthCountry.commonName;
+        }
+
+        if (! _INSEECodeIsForeign(certificationCourseDTO.birthINSEECode)) {
+
+          const candidateBirthCity = await knex('certification-cpf-cities')
+            .select('name', 'INSEECode')
+            .where({ INSEECode: certificationCourseDTO.birthINSEECode })
+            .first();
+
+          certificationCourseDTO.birthplace = candidateBirthCity.name;
+          certificationCourseDTO.birthINSEECode = candidateBirthCity.INSEECode;
+          certificationCourseDTO.birthCountry = 'FRANCE';
+        }
+      }
+
       return _toDomain({ certificationCourseDTO, certificationIssueReportsDTO });
     }
     throw new NotFoundError(`Certification course of id ${certificationCourseId} does not exist.`);
   },
 };
 
+function _INSEECodeIsForeign(INSEECode) {
+  return INSEECode !== FRANCE_INSEE_CODE && _INSECODEStartsWith99(INSEECode);
```

Comment:
```suggestion
  return INSEECode !== FRANCE_INSEE_CODE;
```
C'est suffisant ici pour savoir si c'est un pays étranger ou non

Reply:
Nein, si le code insee est 75115, ce n'est pas un pays étranger :)

Reply:
Arf oui c'est pas faux. Du coup, dans la méthode en dessous, un `INSEECode.startsWith('99')` devrait être mieux que des `charAt`.